mlflow/models/utils.py

```diff
@@ -871,3 +874,53 @@ def get_model_version_from_model_uri(model_uri):
     (name, version) = get_model_name_and_version(client, model_uri)
     model_version = client.get_model_version(name, version)
     return model_version
+
+
+def _enforce_params_schema(params: Optional[Dict[str, Any]], schema: Optional[ParamSchema]):
+    if params in [None, {}] and schema is None:
+        return params
+    if not isinstance(params, dict):
+        raise MlflowException(
+            f"Parameters must be a dictionary. Got type '{type(params).__name__}'."
+        )
+    if not isinstance(schema, ParamSchema):
+        raise MlflowException(
+            "Parameters schema must be an instance of ParamSchema. "
+            f"Got type '{type(schema).__name__}'."
+        )
+    if any(not isinstance(k, str) for k in params.keys()):
+        _logger.warning(
+            "Keys in parameters should be of type `str`, but received non-string keys."
+            "Converting all keys to string..."
+        )
+        params = {str(k): v for k, v in params.items()}
+
+    allowed_keys = {param.name for param in schema.params}
+    ignored_keys = set(params) - allowed_keys
+    if ignored_keys:
+        _logger.warning(
+            f"Invalid arguments {list(ignored_keys)} are ignored for inference. "
+            f"Supported arguments are: {allowed_keys}. "
+            "To enable them, please add corresponding schema in ModelSignature."
+        )
+
+    params = {k: params[k] for k in params if k in allowed_keys}
+
+    invalid_params = set()
+    for param_spec in schema.params:
+        if param_spec.name in params:
+            param_value = params[param_spec.name]
+            try:
+                param_value = ParamSpec.validate_param_spec(param_value, param_spec)
+            except MlflowException as e:
+                invalid_params.add((param_spec.name, e.message))
+        else:
+            params[param_spec.name] = param_spec.default
+
+    if len(invalid_params) > 0:
```

Comment:
We can probably just use the boolean empty check, right?

```suggestion
    if invalid_params:
```